### PR TITLE
feature: Update for rcnet-v2-phase-2

### PR DESCRIPTION
**Pull request by dhedey**
Update for `release/rcnet-v2-phase-2` which includes a few changes in the engine:

```
c9acc419c Merge pull request #1239 from radixdlt/bugfix/fix-well-known-types
a1547f052 Merge pull request #1237 from radixdlt/feature/remove-schema-path
3337e4f3f Merge pull request #1227 from radixdlt/architecture/locked-fields
6bc153855 Merge pull request #1234 from radixdlt/refactor/clean-up-vault-take-by-amount
And the preview panic fix
```

Notably the locked fields was quite a major change, requiring quite a few changes in the node - but thankfully I'd pre-empted the change, so it got absorbed into the `is_locked` field without any changes to the schema.

## Note for integrators:
The NonFungibleData now has to be a tuple, and so the NonFungibleSchema to use `Unit` (ie empty tuple) metadata is now:
```
Tuple(Tuple(Array<Enum>(), Array<Tuple>(), Array<Enum>()), Enum<0u8>(66u8), Array<String>())
```
The 66u8 is the well known type id for "Unit".

### CODE REVIEW

**Review by jakrawcz-rdx — APPROVED** [review]
the "Core API field substate conversion" macro is black magic to me; but other than that it is surprisingly easy update 👍

**Review by malexandru-rdx — COMMENTED** [review]
Typescript SDK needs regeneration too.

File: core-rust/core-api-server/src/core_api/conversions/substates/mod.rs [hunk 1]
```diff
@@ -37,18 +38,59 @@ macro_rules! field_substate {
...
+                    Box::new(models::[<$substate_type Value>] $fields)
+                }
             }
         }
     };
 }
 pub(crate) use field_substate;
 
+// TODO - remove this when TypeInfo is properly wrapped in a FieldSubstate<..> wrapper
+macro_rules! fake_field_substate {
```
[the remark this answers is not in the document]
Comment (dhedey) [reply core-rust/core-api-server/src/core_api/conversions/substates/mod.rs:66]: Possibly - my thinking was that I wanted to keep the invocation site as neat as possible, but I guess that depends how familiar one is with macros hah

[the remark this answers is not in the document]
Comment (dhedey) [reply core-rust/core-api-server/src/core_api/conversions/substates/mod.rs:66]: Happy to change this is you'd rather it be inlined @jakrawcz-rdx ? :)

[the remark this answers is not in the document]
Comment (jakrawcz-rdx) [reply core-rust/core-api-server/src/core_api/conversions/substates/mod.rs:66]: no need, if it is really going to be removed soon anyway 👍

**Pull request merged**